### PR TITLE
Don't raise in `Env`

### PR DESCRIPTION
**Pull request by Julow**
This PR removes exceptions from `Env`. Rebased on https://github.com/jonludlam/odoc/pull/19

`Link` and `Compile` modules are next.

### CODE REVIEW

File: src/xref2/env.ml [hunk 1]
```diff
@@ -431,25 +420,20 @@ let lookup_root_module name env =
...
-      | Some (Resolved (_, m)) -> m
-      | Some (Forward) ->
-        Format.fprintf Format.err_formatter "Forward!\n%!";
-        raise (MyFailure ((identifier :> Odoc_model.Paths.Identifier.t), env))
-      | None ->
-        Format.fprintf Format.err_formatter "None\n%!";
-        raise (MyFailure ((identifier :> Odoc_model.Paths.Identifier.t), env)))
-    | _ -> 
-      Format.fprintf Format.err_formatter "Non root: %a\n%!" Component.Fmt.model_identifier (identifier :> Odoc_model.Paths.Identifier.t);
-      raise (MyFailure ((identifier :> Odoc_model.Paths.Identifier.t), env))
```
Comment (Julow) [inline src/xref2/env.ml:452]: The prints are gone.

File: src/xref2/env.ml [hunk 2]
```diff
@@ -591,24 +575,24 @@ let add_functor_args : Odoc_model.Paths.Identifier.Signature.t -> t -> t =
     in
     match id with
     | (`Module _ | `Result _ | `Parameter _) as mid -> (
-        let m = lookup_module mid env in
-        match m.Component.Module.type_ with
-        | Alias _ -> env
-        | ModuleType e ->
+        match lookup_module mid env with
+        | None -> None
```
Comment (Julow) [inline src/xref2/env.ml:579]: Should we ignore the lookup failure here and simply return `env` ?

Comment (jonludlam) [reply src/xref2/env.ml:579]: Not yet, I think returning `None` is right here - we can choose in `compile.ml` to carry on anyway if we're not in strict mode.

**Comment by Julow** [comment]
Just rebased, the tests are passing.

**Pull request merged**